terraform/vpc.tf

```diff
@@ -5,11 +5,12 @@ module "vpc" {
   source  = "terraform-aws-modules/vpc/aws"
   version = "3.11.0"
 
-  name                 = "vpc"
-  cidr                 = "10.0.0.0/16"
-  azs                  = data.aws_availability_zones.available.names
-  private_subnets      = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
-  public_subnets       = ["10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24"]
+  name = "vpc"
+  cidr = "10.0.0.0/16"
+  azs  = data.aws_availability_zones.available.names
+  # Generate 6 non-overlapping subnets for our VPC. This results in 2^(32-20)=2^12=4096 IPs per subnet.
+  private_subnets      = ["10.0.0.0/20", "10.0.16.0/20", "10.0.32.0/20"]
+  public_subnets       = ["10.0.48.0/20", "10.0.64.0/20", "10.0.80.0/20"]
```

Comment:
Why do we have 3 private & public subnets and what are we using them for

Reply:
Wondering how the values were chosen? Are these values standardized

Reply:
@joyliu-q Honestly, we have 3 public and private subnets just because the [terraform module](https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/latest) we're using. It's so that we have one public and one private subnet per availability zone in AWS (a, b, and c for us).

@jonathankao97 The values are chosen so that the subnets don't overlap. You can run the numbers through a [CIDR calculator](https://www.subnet-calculator.com/cidr.php) where an IP of `10.0.0.0` with a subnet mask of `/20` generates a subnet with the range `10.0.0.0` to `10.0.15.255`. Because of that we choose the next subnet to start at the next available IP `10.0.16.0`